### ISSUE TITLE
Stackable Animations

### ISSUE THREAD

**Issue by davesmith00000** [PurpleKingdomGames/indigo#410]
@hobnob has put some lovely new syntax into the in-flight [roguelike demo](https://github.com/davesmith00000/roguelike) that allows you do do things like this (made up, illustrative):

```scala
myText.moveTo(x, y).fadeIn(5 seconds)
``` 

But if you did:

```scala
myText.moveTo(x, y).fadeIn(5.seconds, runningTime).fadeOut(5.seconds, runningTime)
``` 

Both those animations would action in parallel, we have no way to stack animations so that they run sequentially.

I did bodge something together with `Signal`s many moons ago for a different demo, but it would be nice to have some sort of official support with nice syntax.

**Comment by davesmith00000** [comment]
This will be out in the next release as a feature called `Timeline`s.
